climada/hazard/tc_tracks_synth.py

```diff
@@ -166,6 +206,170 @@ def _one_rnd_walk(track, ens_size, ens_amp0, ens_amp, max_angle, rnd_vec):
+
+    Returns initial bearing (also called forward azimuth) of the n-1 great circle
+    paths define by n consecutive longitude/latitude points. The bearing is the angle
+    (clockwise from North) which if followed in a straight line along a great-circle
+    arc will take you from the start point to the end point. See also:
+    http://www.movable-type.co.uk/scripts/latlong.html
+    Here, the bearing of each pair of consecutive points is computed.
+
+    Parameters:
+        lon (numpy.ndarray of shape (n,)): longitude coordinates, in decimal degrees
+        lat (numpy.ndarray of shape (n,)): latitude coordinates, in decimal degrees
+
+    Returns:
+        bearing_angle (numpy.ndarray of shape (n-1,), in decimal degrees
+    """
+    lon, lat = map(np.radians, [lon, lat])
+    lat_1 = lat[:-1]
+    lon_1 = lon[:-1]
+    lat_2 = lat[1:]
+    lon_2 = lon[1:]
+    delta_lon = lon_2 - lon_1
+    # what to do with the points that don't move?
+    #   i.e. where lat_2=lat_1 and lon_2=lon_1? The angle does not matter in
+    # that case because angular distance will be 0.
+    earth_ang_fix = np.arctan2(np.sin(delta_lon) * np.cos(lat_2),
+                               np.cos(lat_1) * np.sin(lat_2) - \
+                               np.sin(lat_1) * np.cos(lat_2) * np.cos(delta_lon))
+    return np.degrees(earth_ang_fix)
+
+
+@jit
+def _get_angular_distance(lon, lat):
+    """Compute the angular distance of great circle paths defined by consecutive points
+
+    Uses the haversine formulat to get the great circle distance between each pair
+    of consecutive points, in decimal degrees.
+
+    Parameters:
+        lon (numpy.ndarray of shape (n,)): longitude coordinates, in decimal degrees
+        lat (numpy.ndarray of shape (n,)): latitude coordinates, in decimal degrees
+
+    Returns:
+        Angular distance (numpy.ndarray of shape (n-1,), in decimal degrees
+    """
+
+    lon, lat = map(np.radians, [lon, lat])
+    lat_1 = lat[:-1]
+    lat_2 = lat[1:]
+    delta_lon = lon[1:] - lon[:-1]
+    delta_lat = lat_2 - lat_1
+    a = np.sin(delta_lat / 2.0) ** 2 + np.cos(lat_1) * np.cos(lat_2) * np.sin(delta_lon / 2.0) ** 2
+    c = np.degrees(2 * np.arctan2(np.sqrt(a), np.sqrt(1 - a)))
+    return c
+
+
+@jit
+def _get_destination_points(lon, lat, bearing, angular_distance):
+    """Get coordinates of endpoints from a given locations with the provided bearing and distance
+
+    Parameters:
+        lon (numpy.ndarray of shape (n,)): longitude coordinates of start location,
+            in decimal degrees
+        lat (numpy.ndarray of shape (n,)): latitude coordinates of start location,
+            in decimal degrees
+        bearing (numpy.ndarray of shape (n,)): bearing (direction Northward, clockwise)
+        angular_distance (numpy.ndarray of shape (n,)): angular distance, in decimal degrees
+
+    Returns:
+        Angular distance (numpy.ndarray of shape (n,), in decimal degrees
```

Comment:
Something is wrong with the docstring. The documented return value should be an ndarray, but the function returns a pair of ndarrays. Furthermore, the "Returns" part of the docstring says it returns an angular distance while the one-line-summary of the docstring says it returns coordinates of endpoints.

Reply:
Yes, thanks, fixed in e837bc88958cd756867fef5a7ef8d79571c8981c